Examples/Datadog Sample/main.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import DatadogExporter
+import Foundation
+import OpenTelemetrySdk
+import OpenTelemetryApi
+
+
+let clientKey = ""
+
+let sampleKey = "sampleKey"
+let sampleValue = "sampleValue"
+
+let instrumentationLibraryName = "DatadogExporter"
+let instrumentationLibraryVersion = "semver:0.1.0"
+var instrumentationLibraryInfo = InstrumentationLibraryInfo(name: instrumentationLibraryName, version: instrumentationLibraryVersion)
+
+var tracer: TracerSdk
+tracer = OpenTelemetrySDK.instance.tracerProvider.get(instrumentationName: instrumentationLibraryName, instrumentationVersion: instrumentationLibraryVersion) as! TracerSdk
+
+func simpleSpan() {
+    let span = tracer.spanBuilder(spanName: "SimpleSpan").setSpanKind(spanKind: .client).startSpan()
+    span.setAttribute(key: sampleKey, value: sampleValue)
+    span.addEvent(name: "My event", attributes: ["message": AttributeValue.string("test message")])
+    span.end()
+}
+
+func childSpan() {
+    let span = tracer.spanBuilder(spanName: "parentSpan").setSpanKind(spanKind: .client).startSpan()
+    span.setAttribute(key: sampleKey, value: sampleValue)
+    do {
+        var scope = tracer.withSpan(span)
+        let childSpan = tracer.spanBuilder(spanName: "childSpan").setSpanKind(spanKind: .client).startSpan()
+        do {
+            var childScope = tracer.withSpan(childSpan)
+            childSpan.setAttribute(key: sampleKey, value: sampleValue)
+            childScope.close()
+        }
+        childSpan.end()
+        scope.close()
+    }
+    span.end()
+}
+
+let exporterConfiguration = ExporterConfiguration(
+    serviceName: "Otel exporter Example",
```

Comment:
one suggestion is to take the `servicename` from resources API. See [cnvention](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/README.md#user-content-service:~:text=Logical%20name%20of%20the%20service.%20%5B1%5D) and example [in this post](https://medium.com/opentelemetry/opentelemetry-beyond-getting-started-5ac43cd0fe26#e300:~:text=%7B%20%E2%80%9Cservice.name%E2%80%9D%2C%20%E2%80%9CMyBackEndApp%E2%80%9D%20%7D%2C)

Reply:
That's a good idea, but after trying it it looks like the api for modifying the resources is not very functional. I will add it as an issue to the project for a future fix.